src/main/java/com/github/fge/jsonpatch/JsonPathParser.java

```diff
@@ -6,12 +6,43 @@
     private static final String ARRAY_ELEMENT_LAST_REGEX = "\\.(\\d+)$";
 
     public static String tmfStringToJsonPath(String path) {
+        if (!path.startsWith("/") && !path.isEmpty()) {
+            return "$." + path;
+        }
         if ("/".equals(path)) {
             return "$";
         }
-        final String jsonPath = "$" + path.replace('/', '.')
+        final String[] pointerAndQuery = path
+                .replaceAll("(\\w)\\?", "$1#")
+                .split("#", -1);
+        if (pointerAndQuery.length > 2) {
+            // TODO use different exception
```

Comment:
when this `TODO` will be resolved

Reply:
Immediately :D used JsonPatchException here, with a new custom message, thanks